x-pack/plugins/gis/index.js

```diff
@@ -44,7 +46,15 @@ export function gis(kibana) {
       if (gisEnabled) {
         const thisPlugin = this;
         const xpackMainPlugin = server.plugins.xpack_main;
-        initRoutes(server);
+        let routesInitialized = false;
+
+        watchStatusAndLicenseToInitialize(xpackMainPlugin, thisPlugin,
+          async license => {
+            if (license && license.gis && !routesInitialized) {
+              routesInitialized = true;
+              initRoutes(server, license.uid);
```

Comment:
Does it matter that if the license changes and the uid changes that the EMS client will still use the old uid?

Reply:
It looks like up until this point we've only added query params (such as license info), but not so much modified or removed them. I believe this should fix the current issue but should we create another issue to allow updateable query params @thomasneirynck?